contracts/UniswapV3Staker.sol

```diff
@@ -292,22 +295,32 @@ contract UniswapV3Staker is
             SafeMath.sub(incentives[incentiveId].totalRewardUnclaimed, reward)
         );
 
-        /* TODO: This will be fixed in https://github.com/omarish/uniswap-v3-staker/issues/38
-        so that collecting and unstaking are two separate functions */
-        try
-            IERC20Minimal(incentives[incentiveId].rewardToken).transfer(
-                params.to,
-                reward
-            )
-        returns (bool) {
-            // It didn't fail
-        } catch {
-            // It failed
-        }
+        rewards[incentives[incentiveId].rewardToken][msg.sender] = uint128(
+            SafeMath.add(rewards[incentives[incentiveId].rewardToken][msg.sender], reward)
```

Comment:
Every time we call `incentives[incentiveId]` we're querying storage, which can be expensive. A fun experiment, might be to load the incentive into memory here inside the function , and use the memory instance for every time we query it. Then see how that affects the gas costs. 

(less important: could also experiment with writing to the one and memory then saving it all to storage at the end to see how that affects gas)

Reply:
Just added, nice gas savings :)